client/src/components/DatasetInformation/DatasetError.vue

```diff
@@ -61,30 +61,31 @@
                         show
                         >{{ resultMessage[0] }}</b-alert
                     >
-                    <div v-if="showForm" id="fieldsAndButton">
-                        <span class="mr-2 font-weight-bold">{{ emailTitle | l }}</span>
-                        <span>{{ currentUserEmail }}</span>
-                        <FormElement
-                            id="dataset-error-message"
-                            v-model="message"
-                            :area="true"
-                            title="Please provide detailed information on the activities leading to this issue:" />
-                        <b-button
-                            id="dataset-error-submit"
-                            variant="primary"
-                            class="mt-3"
-                            @click="submit(dataset, jobDetails.user_email)">
-                            <font-awesome-icon icon="bug" class="mr-1" />Report
-                        </b-button>
-                    </div>
+                    <CurrentUser v-slot="{ user }">
```

Comment:
In general we don't want to move code *to* provider-style data access -- what wasn't working correctly with the getter?

Reply:
@dannon [That commit I made](https://github.com/galaxyproject/galaxy/pull/15042/commits/378d61c115337a8cf5c34bf859f57cc58148dd99) was solely related to getting the Jest test to pass. On a macro-level previously, there was no User Email being sent to the front-end (the email address field wasn't getting pre-filled even). So, I needed a way to get User object information onto the page but that also seemed to fit our preferred convention. 
Alternatives I explored included: (1) using `this.currentUser.email` which resulted in "Namespace Map" errors from the `yarn jest watch` front-end tests and caused all existing tests for this component to fail (2) using `Galaxy.user.email` in the `router.js` file which was a very clean implementation but was suggested by @guerler not to do this "older" method. I am open to other ideas...however, perhaps it is the Jest test that needs to be fixed in this instance?

Reply:
Yes, we want to avoid using the global `Galaxy` object wherever we can. In principle the provider uses the store getter, so they should be functionally equivalent, but as mentioned before, moving forward and once converted to the composition api, we'll use composables.

Reply:
@dannon Thanks for that clarity - will do that moving forward. One thing I forgot to add was the message when a user isn't authenticated as stated in your previous [comment here](https://github.com/galaxyproject/galaxy/issues/13894#issuecomment-1123706051). I will push that change shortly and reply back to this thread so you know its updated.

Reply:
@dannon Updated with additional screenshot image in this PR's Description field.